src/main/java/duke/command/CommandParser.java

```diff
@@ -0,0 +1,142 @@
+package duke.command;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.exception.DateParseException;
+import duke.exception.IncompleteTaskException;
+import duke.exception.InvalidSearchException;
+import duke.exception.InvalidTaskException;
+import duke.exception.UnknownCommandException;
+import duke.task.TaskType;
+
+/**
+ * CommandParser is a class that parses input from the user, and returns a relevant command.
+ */
+public class CommandParser {
+    /**
+     * Processes String value commands given by the user and returns the relevant command.
+     * @param command A String value which is the user input.
+     * @return A Command corresponding to user input.
+     * @throws UnknownCommandException if the Command type cannot be determined from user input.
+     * @throws DateParseException if the user input contains a date object that cannot be parsed.
+     * @throws IncompleteTaskException if user input contains incorrect or insufficient information for task creation.
+     * @throws InvalidTaskException if task index provided by user is invalid or missing.
+     */
+    public static Command parseCommand(String command) throws UnknownCommandException, DateParseException,
+            IncompleteTaskException, InvalidTaskException, InvalidSearchException {
+        command = command.trim();
+        if (command.equals("bye")) {
+            return new ExitCommand();
+        } else if (command.equals("list")) {
+            return new ListCommand();
+        } else if (command.equals("today")) {
+            return new TodayCommand();
+        } else if (command.length() >= 4 && command.substring(0, 4).equals("find")) {
+            return parseFindCommand(command);
+        } else if (command.length() >= 4 && command.substring(0, 4).equals("done")) {
+            return parseDoneCommand(command);
+        } else if (command.length() >= 6 && command.substring(0, 6).equals("delete")) {
+            return parseDeleteCommand(command);
+        } else if (validAddTaskCommand(command)) {
+            return parseAddTaskCommand(command);
+        } else {
+            throw new UnknownCommandException("Oh noes! I'm not sure what that means ;A;");
+        }
```

Comment:
consider changing theses if-else to a switch statement for clarirty

Reply:
My original plan was to use a switch statement, but that meant I couldn't call other methods to check the validity of the command or perform operations like command.split(" ")[0].equals("done"), so I settled for if-else instead. Thanks for the feedback though!